lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -514,7 +515,11 @@ export class UserAgentApplication {
             }
         }
 
-        acquireTokenAuthority.resolveEndpointsAsync(this.telemetryManager, request.correlationId).then(async () => {
+        try {
+            if (!acquireTokenAuthority.validateTenantDiscoveryResponse()) {
+                await AuthorityFactory.resolveAuthorityAsync(serverAuthenticationRequest.authorityInstance);
+            }
```

Comment:
Is there an else here?

Reply:
Nothing functional, probably good to add logger calls.